openedx/core/djangoapps/content/course_overviews/signals.py

```diff
@@ -13,7 +14,9 @@ def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable
     Catches the signal that a course has been published in Studio and
     updates the corresponding CourseOverview cache entry.
     """
-    CourseOverview.load_from_module_store(course_key)
+    previous_course_overview = CourseOverview.get_from_ids_if_exists([course_key]).get(course_key)
+    updated_course_overview = CourseOverview.load_from_module_store(course_key)
+    _check_for_course_changes(previous_course_overview, updated_course_overview)
```

Comment:
Could you just ride on the standard django signal for a model object (`CourseOverview`, in this case) being updated? I'm pretty sure that signal gives you a before and after comparison.

Reply:
An advantage to that is that you could get rid of the specific `COURSE_PACING_CHANGED` and `COURSE_START_DATE_CHANGED` signals, and just have the code that needs to listen listen directly to the model changing.

Reply:
Good point.  However, it looks like [post_save](https://docs.djangoproject.com/en/1.11/ref/signals/#post-save) doesn't provide both the before and after model-instances.  And we don't want to fire possibly-expensive tasks during `pre_save` to avoid extra work in the event of an exception.

So for now I'm going to leave it as is.  If we find this pattern doesn't scale, we can consider introduce a generic `COURSE_OVERVIEW_CHANGED` signal that provides both previous and updated values..